src/interfaces/transaction.ts

```diff
@@ -79,8 +79,8 @@ export interface Transaction {
   /** Internally creates Raw Transaction data.
    *  Requires at least one action set. Must be called before sign() */
   prepareToBeSigned(): Promise<void>
-  /** Whether the transaction needs stakes to execute on chain */
-  requiresResources: boolean
+  /** Whether a transaction requires chain resources to execute */
+  supportsResources: boolean
```

Comment:
@traylewin this change conflicts from its [spec](https://docs.google.com/document/d/1OghaLQ5q_ij__znn07q7W0C-WJEp4wwEddxJK-iB67A/edit#bookmark=id.a7i10x86rqh6)

Reply:
This is the pattern we use for optional features so I switched to that